pkg/services/preference/prefimpl/pref_store.go

```diff
@@ -0,0 +1,119 @@
+package prefimpl
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	pref "github.com/grafana/grafana/pkg/services/preference"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/services/sqlstore/db"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type store interface {
+	Get(context.Context, *pref.GetPreferenceQuery) (*pref.Preference, error)
+	GetDefaults() *pref.Preference
+	List(context.Context, *pref.ListPreferenceQuery) ([]*pref.Preference, error)
+	Set(context.Context, *pref.SavePreferenceCommand) (*pref.Preference, error)
+}
+
+type sqlStore struct {
+	db  db.DB
+	cfg setting.Cfg
+}
+
+func (s *sqlStore) Get(ctx context.Context, query *pref.GetPreferenceQuery) (*pref.Preference, error) {
+	var prefs pref.Preference
+	err := s.db.WithDbSession(ctx, func(sess *sqlstore.DBSession) error {
+		_, err := sess.Where("org_id=? AND user_id=? AND team_id=?", query.OrgID, query.UserID, query.TeamID).Get(&prefs)
+
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+	return &prefs, err
+}
+
+func (s *sqlStore) GetDefaults() *pref.Preference {
```

Comment:
I don't think this method should be part of the store at all

Reply:
Also, I believe removing it will disconnect store from the cfg

Reply:
It doesn't belong to store, I can move it to the service or somewhere else?